File: erp-frontend/src/app/core/catalogos/ubicacion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { API_URL } from '../api-config';
import { UbicacionRequest, UbicacionResponse } from './ubicacion';

const BASE_URL = `${API_URL}/ubicaciones`;

/** Salones/direcciones de eventos (propios o de terceros). Usado por Cotizaciones y Eventos. */
@Injectable({ providedIn: 'root' })
export class UbicacionService {
  constructor(private readonly http: HttpClient) {}

  listar(): Observable<UbicacionResponse[]> {
    return this.http.get<UbicacionResponse[]>(BASE_URL);
  }

  crear(request: UbicacionRequest): Observable<UbicacionResponse> {
    return this.http.post<UbicacionResponse>(BASE_URL, request);
  }
}

File: erp-backend/src/main/java/com/lacasadelchef/erp/ubicacion/UbicacionService.java
package com.lacasadelchef.erp.ubicacion;

import com.lacasadelchef.erp.ubicacion.dto.UbicacionRequest;
import com.lacasadelchef.erp.ubicacion.dto.UbicacionResponse;

import java.util.List;

public interface UbicacionService {

    List<UbicacionResponse> listar();

    UbicacionResponse obtenerPorId(Integer id);

    UbicacionResponse crear(UbicacionRequest request);

    UbicacionResponse actualizar(Integer id, UbicacionRequest request);

    void eliminar(Integer id);
}
